sigma/filters.py

```diff
@@ -0,0 +1,210 @@
+import random
+import re
+import string
+from dataclasses import dataclass, field
+from typing import List, Optional, Union
+from uuid import UUID
+
+import yaml
+
+from sigma import exceptions as sigma_exceptions
+from sigma.correlations import SigmaCorrelationRule
+from sigma.exceptions import SigmaRuleLocation
+from sigma.rule import (
+    SigmaYAMLLoader,
+    SigmaLogSource,
+    SigmaDetections,
+    SigmaDetection,
+    SigmaRule,
+    SigmaRuleBase,
+)
+
+
+class SigmaFilterLocation(sigma_exceptions.SigmaRuleLocation):
+    """Location of Sigma filter in source file."""
+
+    pass
+
+
+@dataclass
+class SigmaGlobalFilter(SigmaDetections):
+    rules: List[UUID] = field(default_factory=list)
```

Comment:
I ended up using it but it required a local import for SigmaRuleCollection because that's where the referencing is. Can split it out into a "SigmaRuleReferenceResolver" class later if required for PR.